multimodal/src/autogluon/multimodal/learners/default_learner.py

```diff
@@ -0,0 +1,1724 @@
+        self,
+        label: Optional[str] = None,
+        problem_type: Optional[str] = None,
+        column_types: Optional[dict] = None,
+        # query: Optional[Union[str, List[str]]] = None,
+        # response: Optional[Union[str, List[str]]] = None,
+        # match_label: Optional[Union[int, str]] = None,
+        # pipeline: Optional[str] = None,
+        # presets: Optional[str] = None,
+        eval_metric: Optional[str] = None,
+        path: Optional[str] = None,
+        verbosity: Optional[int] = 2,
+        num_classes: Optional[int] = None,
+        classes: Optional[list] = None,
+        enable_progress_bar: Optional[bool] = None,
+    ):
+        super().__init__()
+
+        self._label_column = label
+        self._problem_type = problem_type
+        # self._presets = presets.lower() if presets else None  # NOTE: This will be handled by predictor.
+        self._eval_metric_name = eval_metric
+        self._validation_metric_name = None
+        self._output_shape = num_classes
+        self._classes = classes  # TODO: To be refactored into detection learner only
+
+        self._ckpt_path = None
+        self._pretrained_path = None
+        self._config = None
+        self._df_preprocessor = None  # initialize it as None.
+        self._column_types = column_types  # predictor will setup column_types, and set values here.
+        self._data_processors = None  # initialize it as None.
+        self._model_postprocess_fn = None
+        self._model = None
+        # self._resume = False  # This is for predictor to handle.
+        self._verbosity = verbosity
+        # self._warn_if_exist = warn_if_exist  # unused
+        self._enable_progress_bar = enable_progress_bar if enable_progress_bar is not None else True
+        # self._init_scratch = init_scratch  # NOTE: is this for predictor to handle?
+        # self._sample_data_path = sample_data_path  # this is for detection problem only
+        self._fit_called = False  # While using ddp, after fit called, we can only use single gpu.
+        self._save_path = path
+
+        # Summary statistics used in fit summary. TODO: wrap it in a class.
+        # self._total_train_time = None
+        self._best_score = None
+
+    @override
+    def path(self) -> str | None:
+        return self._save_path
+
+    @override
+    def label(self) -> str | None:
+        return self._label_column
+
+    @property
+    @override
+    def problem_type(self) -> str | None:
+        return self._problem_type
+
+    @property
+    @override
+    def problem_property(self) -> ProblemTypeProperty | None:
+        if self._problem_type is None:
+            return None
+        else:
+            return PROBLEM_TYPES_REG.get(self._problem_type)
+
+    @property
+    @override
+    def class_labels(self) -> list | None:
+        return self._classes
+
+    @property
+    @override
+    def positive_class(self) -> int | None:
+        return None
+
+    @override
+    def fit(
```

Comment:
docstring?

Reply:
+1, please add sequence flow for call stacks as well

Reply:
Consider the following call back functions:

```
_setup_save_path()
_infer_problem_type()
_split_train_tuning()
_infer_column_types()
_infer_output_shape()
_infer_metric()
_update_hyperparameters()
_split_hyperparameters()
_get_config()
_get_df_preprocessor()
_get_data_processors()
_get_torchmetric()
_get_loss_func()
_prepare_for_distillation()
_get_data_module()
_get_lightning_module()
_get_lightning_callbacks()
_get_lightning_plugins()
_get_tf_logger()
_init_lightning_trainer()
_run_lightning_trainer()
_fit_postprocess()
```